packages/coding-standards/composer.json

```diff
@@ -22,12 +22,12 @@
     },
     "autoload": {
         "psr-4": {
-            "Shopsys\\CodingStandards\\": "src"
+            "Shopsys\\CodingStandards\\": "src/"
         }
     },
     "autoload-dev": {
         "psr-4": {
-            "Shopsys\\Tests\\Unit\\CodingStandards\\": "tests"
+            "Tests\\CodingStandards\\": "tests/"
```

Comment:
What if there will be 3rd party with the same generic `Tests\CodingStandards` namespace? 

Standard unique naming prevents these bugs:

- Vendor\Package: src
- Vendor\Package\Tests: tests

Reply:
It's in `autoload-dev`, so it shouldn't be a problem. Test classes of 3rd party packages should not be autoloaded.

We have adapted this standard while upgrading to Symfony 3.4. 
https://symfony.com/doc/3.4/testing.html

In 4.0 they switched from `Tests\Namespace` to `Vendor\Tests\Namespace`.
https://symfony.com/doc/current/testing.html

We have never run into issues with this approach but we will keep in mind that it could happen.

Reply:
> It's in autoload-dev...

It's very easy to confuse those and use non-vendor-prefixed naming in `autoload`.

> We have adapted this standard while upgrading to Symfony 3.4.
https://symfony.com/doc/3.4/testing.html

Symfony documentation is not very good for testing practise. The source code, on the other hand, tells the true story. I learned unique class naming (test or not) from there. They've been doing for ages, e.g. see Symfony 2.8: https://github.com/symfony/symfony/blob/v2.8.0/src/Symfony/Component/Console/Tests/ApplicationTest.php#L12

> We have never run into issues with this approach but we will keep in mind that it could happen.

This just opens the door to such bugs.

I have run into couple similar issues in my life and it's really exhausting to find such bug, moreover when you really on standards and best practice of open-source.

Reply:
This is my favorite example of "unique" approach that caused already so many troubles related to phpunit, diff and constraint invalid version r apis: https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/2.11/composer.json#L65-L80

Reply:
Good points, I have created a user story to solve this. Thanks for pointing this out.